### ISSUE TITLE
Need check_val_every_n_steps in Trainer

### ISSUE THREAD

**Issue by del2z** [PyTorchLightning/pytorch-lightning#5565]
## 🚀 Feature
Add an argument `check_val_every_n_steps` in `Trainer.__init__` function to check metrics of validation set for certain steps.

### Motivation
For many tasks, large models are trained in steps not complete epochs, especially pretrained models in CV and NLP. As a consequence, step-based arguments like `max_steps`, `log_every_n_steps` may be more convenient than epoch-based ones. However, the `Trainer` API only has a `check_val_every_n_epoch` argument for computing metrics of validation data. It's very helpful to have an additional argument like `check_val_every_n_steps` in `Trainer` constructor.

### Pitch
Trainer.__init__(..., check_val_every_n_epoch=1, check_val_every_n_steps=100, ...)

**Comment by del2z** [comment]
Another confusing concept is `batch_idx` in `training_step`, `validation_step` and `test_step`. A detailed example or illustration may be helpful to understand this concept. From my experience, `batch_idx` may be not widely used in developing models.

**Comment by rohitgr7** [comment]
there is `val_check_interval` for that.

**Comment by yuvalkirstain** [comment]
@rohitgr7 val_check_interval can't exceed a single epoch. So it does not support evaluation every `n` steps where `n` is larger than the amount of batches in the dataloader.

**Comment by rohitgr7** [comment]
hey @yuvalkirstain not yet.
here is the tracking issue: https://github.com/PyTorchLightning/pytorch-lightning/issues/8135
